### PR TITLE
Homework7: Aleksandr Levdansky

### PR DESCRIPTION
**Pull request by Coromilly**

### CODE REVIEW

File: task1_3.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+try-except block.
+"""
+
+
+def add_to_list_in_dict(thedict, listname, element):
+    try:
+        try:
+            lst = thedict[listname]
+        except TypeError:
+            raise TypeError("Input correct arguments.")
```
Comment (belovmd) [inline task1_3.py]: you can print error and just raise without anything

File: task1_3.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        print("%s already has %d elements." % (listname, len(lst)))
+    except KeyError:
+        thedict[listname] = []
```
Comment (belovmd) [inline task1_3.py]: ...] = [element]?

File: task3.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"""Cover function from 1.3 with tests."""
+from task1_3 import add_to_list_in_dict
+import unittest
+
+
+class FuncTest(unittest.TestCase):
+    def setUp(self):
```
Comment (belovmd) [inline task3.py]: it's strange we put all these constants to setup but use in different tests different values. 
usually test is setup, action, assert. your one is just equal

File: task2.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+def count_characters(s):
+    """Function count and return the numbers of each character in a string."""
+    dct = {}
+    for symb in s:
+        dct[symb] = dct.get(symb, 0) + 1
+    return dct
+
+
+class TestCountCharacters(unittest.TestCase):
```
Comment (belovmd) [inline task2.py:13]: why in count characters test collection are upper lower etc?

File: task2.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    def test_normal(self):
```
Comment (belovmd) [inline task2.py:14]: we need more tests for this function

**Review by belovmd — COMMENTED** [review]
please address these issues

File: task1_3.py (hunk 1 above)
Comment (Coromilly) [reply task1_3.py]: solved.

File: task1_3.py (hunk 2 above)
Comment (Coromilly) [reply task1_3.py]: solved.

File: task3.py (hunk 3 above)
Comment (Coromilly) [reply task3.py]: solved.

File: task2.py (hunk 4 above)
Comment (Coromilly) [reply task2.py:13]: solved.

File: task2.py (hunk 5 above)
Comment (Coromilly) [reply task2.py:14]: solved.

File: task3.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+"""Cover function from 1.3 with tests."""
+from task1_3 import add_to_list_in_dict
+import unittest
+
+
+class FuncTest(unittest.TestCase):
```
Comment (belovmd) [inline task3.py:6]: why here are only string keys?

File: task2.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        res = count_characters('coromilly')
+        self.assertEqual(
+            res, {'c': 1, 'o': 2, 'r': 1, 'm': 1, 'i': 1, 'l': 2, 'y': 1})
+
+    def test_default_value(self):
```
Comment (belovmd) [inline task2.py:19]: it has strange default value. i.e. we can call it with empty args and check assert raises

**Review by belovmd — COMMENTED** [review]
8,5

**Pull request closed without merging**